sdk/go/common/workspace/project.go

```diff
@@ -965,3 +966,32 @@ func save(path string, value interface{}, mkDirAll bool) error {
 	//nolint:gosec
 	return os.WriteFile(path, b, 0o644)
 }
+
+// To mitigate an import cycle, we define this here.
+const PulumiTagsConfigKey = "pulumi:tags"
+
+// SetProjectTagsConfig sets the project tags config to the given map of tags.
+func (proj *Project) SetProjectTagsConfig(tags map[string]string) {
```

Comment:
Add/Append/Merge rather than Set. This isn't setting the whole map it's adding the new entries in.